doc/helpers/responsive-extension.md

```diff
@@ -0,0 +1,109 @@
+---
+uid: Toolkit.UI.ResponsiveExtension
+---
+# ResponsiveExtension
+
+## Summary
+The `ResponsiveExtension` class is a markup extension in the Uno Toolkit UI library that enables the adaptation of UI element properties based on the current window width.
+This functionality provides a dynamic and responsive user interface experience.
+
+### Inheritance 
+Object &#8594; MarkupExtension &#8594; ResponsiveExtension
+
+### Constructors
+| Constructor           | Description                                                    |
+|-----------------------|----------------------------------------------------------------|
+| ResponsiveExtension() | Initializes a new instance of the `ResponsiveExtension` class. |
+
+### Properties
+| Property   | Type             | Description                                                |
+| ---------- | ---------------- | ---------------------------------------------------------- |
+| Narrowest  | object           | Value to be used when the screen size is at its narrowest. |
+| Narrow     | object           | Value to be used when the screen size is narrow.           |
+| Normal     | object           | Value to be used when the screen size is normal.           |
+| Wide       | object           | Value to be used when the screen size is wide.             |
+| Widest     | object           | Value to be used when the screen size is at its widest.    |
+| Layout     | ResponsiveLayout | Overrides the screen size thresholds/breakpoints.          |
+
+### Usage
+```xml
+xmlns:utu="using:Uno.Toolkit.UI"
+...
+<Page.Resources>
+	<SolidColorBrush x:Key="NarrowRed">Crimson</SolidColorBrush>
+	<SolidColorBrush x:Key="WideBlue">Blue</SolidColorBrush>
+</Page.Resources>
+...
+<TextBlock Text="Asd"
+		   Background="{utu:Responsive Narrow={StaticResource NarrowRed}, Wide={StaticResource WideBlue}}" />
+```
+
+## Remarks
+**Property type limitation**: Content property values other than strings must be appropriately typed for the markup extension to interpret them correctly.
+In the usage example above, the values `NarrowRed` and `WideBlue` are properly typed as they refer to the `StaticResource` keys defined in `Page.Resources`.
+For instance, using `Background={utu:Responsive Narrow=SkyBlue, Wide=Pink}` would be incorrect, while the string literal usage under `<TextBlock Text="{utu:Responsive Narrow='Narrow format', Wide='Wide format'}" />` is accepted.
```

Comment:
@Xiaoy312 there's no way around this?

How is MAUI able to do things like this with [OnPlatform](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/customize-ui-appearance?view=net-maui-8.0#customize-ui-appearance-based-on-the-platform)?

```xml
<BoxView Color="{OnIdiom Yellow, Phone=Red, Tablet=Green, Desktop=Blue}"
         WidthRequest="{OnIdiom 100, Phone=200, Tablet=300, Desktop=400}"
         HeightRequest="{OnIdiom 100, Phone=200, Tablet=300, Desktop=400}"
         HorizontalOptions="Center" />
```

Reply:
it is possible, you gotta internally implement type conversion (btw that wont be available to uwp)
with the type hinted by IProvideValueTarget::TargetProperty::Type
however that is not done
we can probably add that in a follow-up pr